Unity.Options/Options.cs

```diff
@@ -868,27 +867,51 @@ private static bool Unprocessed(ICollection<string> extra, Option def, OptionCon
             return false;
         }
 
-        private readonly Regex ValueOption = new Regex(
-                @"^(?<flag>--|-|/)(?<name>[^:=]+)((?<sep>[:=])(?<value>.*))?$");
-
         protected bool GetOptionParts(string argument, out string flag, out string name, out string sep, out string value)
         {
             if (argument == null)
                 throw new ArgumentNullException("argument");
-
-            flag = name = sep = value = null;
-            Match m = ValueOption.Match(argument);
-            if (!m.Success)
-            {
+            
+            flag = null;
+            name = null;
+            sep = null;
+            value = null;
+            
+            if (argument.Length < 2)
                 return false;
-            }
-            flag  = m.Groups["flag"].Value;
-            name  = m.Groups["name"].Value;
-            if (m.Groups["sep"].Success && m.Groups["value"].Success)
+            
+            var startAt = 0;
+            if (argument[0] == '-')
+                startAt++;
+            else
+                return false;
+
+            if (argument[1] == '-')
+                startAt++;
+
+            var sb = new StringBuilder();
+
+            for(int i = startAt; i < argument.Length; i++)
             {
-                sep   = m.Groups["sep"].Value;
-                value = m.Groups["value"].Value;
+                var c = argument[i];
+                if ((c == ':' || c == '=') && sep == null)
+                {
+                    sep = c.ToString();
+                    name = sb.ToString();
+                    sb.Clear();
+                }
+                else
+                {
+                    sb.Append(c);
+                }
             }
+
+            // Bool values can show up as '--bool-value' where there is no sep, in that case the string builder will contain the name instead of the value
+            if (sep == null)
+                name = sb.ToString();
+            else
+                value = sb.ToString();
+            flag = startAt == 1 ? "-" : "--";
```

Comment:
The original regex allows for passing options with a slash like `/option`, which I never actually knew:
`(?<flag>--|-|/)`

I don't know if anyone actually uses that mode/ability anywhere, but probably best to replicate it.